agents/skala-r/fence_skalar.py

```diff
@@ -0,0 +1,226 @@
+    parameters = {
+        "uuid": str(options["--plug"])
+    }
+
+    vm_info = requests.get(url=URL4, params=parameters, verify=ssl_verify, cookies=cookie)
+    jvm_info = vm_info.json()
+    if jvm_info["vm_list"]["items"] == []:
+        raise NameError('Can not find VM by uuid.')
+    logging.debug("VM_INFO:\n{}".format(json.dumps(vm_info.json(), indent=4, sort_keys=True)))
+    status_v = jvm_info["vm_list"]["items"][0]["status"]
+    if status_v not in state:
+        raise Exception('Unknown VM state: {}.'.format(status_v))
+    return state[status_v]
+
+
+def set_power_status(conn, options):
+    action = {
+        "on" : "vm_start",
+        "reboot": "vm_restart",
+        "off" : "vm_stop"
+        }
+
+    vm_id_v = get_vm_id(options["--ip"], options["--plug"], options, cookie)
+    vm_task(options["--ip"], vm_id_v, action[options["--action"]], options, cookie)
+    return
+
+
+def get_list(conn, options):
+    outlets = {}
+    URL5 = proto + options["--ip"] + '/api/0/vm'
+    
+    vm_info = requests.get(url=URL5, verify=ssl_verify, cookies=cookie)
+    jvm_info = vm_info.json()
+    list_jvm = jvm_info["vm_list"]["items"]
+    for elem in list_jvm:
+        outlets[elem["name"]] = (elem["uuid"], None)
+    return outlets
+
+
+def define_new_opts():
+    all_opt["graceful"] = {
+            "getopt" : "",
+            "longopt" : "graceful",
+            "help" : "--graceful                     vm_stop command parameter, graceful stop or not, default false", 
+            "required" : "0",
+            "shortdesc" : "vm_stop command parameter, graceful stop or not, default false",
+            "order" : 1}
+
+    all_opt["force"] = {
+            "getopt" : "",
+            "longopt" : "force",
+            "help" : "--force                        vm_stop command parameter, force stop or not, default false",
+            "required" : "0",
+            "shortdesc" : "vm_stop command parameter, force stop or not, default false", 
+            "order" : 1}
+
+
+def main():
+    global cookie, proto, ssl_verify
+    define_new_opts()
+    device_opt = ["ipaddr", "login", "passwd", "port", "web", "ssl", "verbose", "graceful", "force"]
+    
+    atexit.register(atexit_handler)
+    options = check_input(device_opt, process_input(device_opt))
+
+    docs = {}
+    docs["shortdesc"] = "Skala-R Fence agent"
+    docs["longdesc"] = "A fence agent for Skala-R."
+    docs["vendorurl"] = "https://www.skala-r.ru/"
+    show_docs(options, docs)
+    options["eol"] = "\r"
+    
+    run_delay(options)
+    
+    proto = "https://"
+    if "--ssl" in options or "--ssl-secure" in options:
+        ssl_verify = True
+    elif "--ssl-insecure" in options:
+        ssl_verify = False
+        urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+    else:
+        proto = "http://"
+        ssl_verify = False
+        urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
```

Comment:
You can remove these 2 lines, as they only apply for HTTPS.

Reply:
Generally yes, but if web server redirects to https, requests.Session tries to make a request to https and to this request these params will be useful.

Reply:
Ah perfect.